src/consumer/component/dependencies/dependency-resolver/dependencies-resolver.ts

```diff
@@ -1225,9 +1225,8 @@ either, use the ignore file syntax or change the require statement to have a mod
    */
   _getPackageJson(): Record<string, any> | undefined {
     const componentMap = this.component.componentMap;
-    // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
-    const isAuthor = componentMap.origin === COMPONENT_ORIGINS.AUTHORED;
-    if (isAuthor) {
+    const isAuthor = componentMap?.origin === COMPONENT_ORIGINS.AUTHORED;
+    if (isAuthor && !R.isEmpty(this.consumer.packageJson.packageJsonObject ?? {})) {
```

Comment:
I think in some cases it might be an issue.
Users can have a package.json for just one package. And if it's there, it'll ignore the component-from-model. As a result, the imported component will be modified in the e2e-case. Also, it doesn't take into account the workspace.jsonc.
Maybe these two need to be rewritten according to Harmony:
```
    this.populatePeerPackageDependencies();
    this.applyWorkspacePolicy();
```

Reply:
so I need to remove package.json from getPackageJson (and rename the function)

and I need to read the policies from package.json as well in applyWorkspacePolic()